src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/FilterLongReadAlignmentsSAMSpark.java

```diff
@@ -248,7 +247,10 @@ private static boolean contigFilter(final AlignedContig contig) {
                 .orElseThrow(() -> new GATKException("Cannot find best-scoring configuration on alignments of contig: " + alignedContig.contigName));
 
         return IntStream.range(0, allConfigurations.size())
-                .filter(i -> scores.get(i) >= maxScore)
+                .filter(i -> {
+                    final Double s = scores.get(i);
+                    return s >= maxScore || maxScore - s <= Math.ulp(s); // sometimes, not sure what happened, two configurations with would-be-same-scores have differ by a Math.ulp(appear-to-be-lower)
```

Comment:
I think there's a typo in your comment, maybe you mean "two configurations which should have the same scores differ by a Math.ulp".

I think it might be better to explicitly set a delta for tolerance of equal alignment scores rather than using `Math.ulp`, primarily because I don't quite understand where the difference is coming from and why Math.ulp is guaranteed to fix it. What do you think?

Reply:
i went back and digged a bit deeper.
I think the problem is an inherent sin of comparing floating point numbers that could potentially be very close to each other.
I initially discovered the problem by debugging for an unexpected explosion: two scores computed to be the same by the score computing unit, but when it is sent back to the calling code here, one of them is changed, slightly by the Math.ulp() amount. 

I've added an advanced parameter that defaults to zero (but floating point numbers are never exactly zero I guess), and it is argued [here](http://www.cygnus-software.com/papers/comparingfloats/Comparing%20floating%20point%20numbers.htm) that ULP is probably the least evil way to go.